cmd/fluxctl/root_cmd.go

```diff
@@ -92,6 +92,12 @@ func (opts *rootOpts) Command() *cobra.Command {
 }
 
 func (opts *rootOpts) PersistentPreRunE(cmd *cobra.Command, _ []string) error {
+	// skip port forward for version command
+	switch cmd.Use {
+	case "version":
+		return nil
+	}
+
```

Comment:
I am not familiar with this code; could you please explain how this works? In particular, how does this result in `fluxctl` printing out it's own version _and then_ the `fluxd` version? Which is what #1318 proposes (with the latter step failing if it cannot connect). By my naive reading of the code this change will stop `fluxctl` from connecting to `fluxd` altogether.

Reply:
@dholbach assumed that the `version` command prints the client and server versions (like kubectl does) but in fact fluxctl only prints the client version.

Reply:
If we want the same behaviour like kubectl we could make a new issue for this feature. My PR prevents the `version` command from crashing if fluxd is not accessible.